libraries/log_entry.rb

```diff
@@ -17,14 +17,15 @@ class LogEntry < BaseResource
     resource_name :ilo_log_entry
 
     load_base_properties
-    property :log_type, String, required: true, equal_to: ['IEL', 'IML']
+    property :log_type, String, required: true, equal_to: ['IEL', 'IML', :IEL, :IML]
     property :severity_level, String, equal_to: ['OK', 'Warning', 'Critical']
     property :dump_file, String
     property :owner, [String, Integer], default: ENV['USER'] || ENV['USERNAME']
     property :group, [String, Integer], default: ENV['USER'] || ENV['USERNAME']
     property :duration, Integer, default: 24
 
     action :dump do
+      raise 'Please set the :dump_file property (String)' unless dump_file
```

Comment:
We definitely don't need to, but it just makes the error message a little bit more clear. With it gone, the error is:

```
    ================================================================================
    Error executing action `dump` on resource 'ilo_log_entry[set a subset of BIOS settings]'
    ================================================================================

    ArgumentError
    -------------
    You must supply a name when declaring a file resource

    Cookbook Trace:
    ---------------
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/resource_builder.rb:63:in `build'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:302:in `build_resource'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/dsl/declare_resource.rb:259:in `declare_resource'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/libraries/log_entry.rb:36:in `block in <class:LogEntry>'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `instance_eval'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:132:in `compile_and_converge_action'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
    /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

    Resource Declaration:
    ---------------------
    # In /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/recipes/log_test.rb

      6: ilo_log_entry 'set a subset of BIOS settings' do
      7:   ilos my_ilos
      8:   log_type 'IEL'
      9: end

    Compiled Resource:
    ------------------
    # Declared in /home/vagrant/chef-repo/.chef/local-mode-cache/cache/cookbooks/ilo/recipes/log_test.rb:6:in `from_file'

    ilo_log_entry("dump IEL logs") do
      action [:dump]
      retries 0
      retry_delay 2
      default_guard_interpreter :default
      declared_type :ilo_log_entry
      cookbook_name "ilo"
      recipe_name "log_test"
      log_type "IEL"
    end

    Platform:
    ---------
    x86_64-linux

[2016-09-07T19:21:35+00:00] INFO: Running queued delayed notifications before re-raising exception

Running handlers:
[2016-09-07T19:21:35+00:00] ERROR: Running exception handlers
Running handlers complete
[2016-09-07T19:21:35+00:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 06 seconds
[2016-09-07T19:21:35+00:00] FATAL: Stacktrace dumped to /home/vagrant/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
[2016-09-07T19:21:35+00:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2016-09-07T19:21:35+00:00] ERROR: ilo_log_entry[set a subset of BIOS settings] (ilo::log_test line 6) had an error: ArgumentError: You must supply a name when declaring a file resource
[2016-09-07T19:21:35+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

```

So it's clear which resource it's coming from, but maybe not clear exactly what you need to do. It's not necessary, but I don't think it hurts.